src/main.rs

```diff
@@ -251,8 +270,29 @@ fn run() -> Result<(), Box<::std::error::Error>> {
 
     let stdin = io::stdin();
 
+    let start_offset = if let Some(start) = matches
+        .value_of("start")
+        .and_then(|s| s.parse::<u64>().ok())
+    {
+        start
+    } else {
+        0
+    };
+    let mut print_offset = false;
+    let print_idx = start_offset - (start_offset % 16);
+
     let mut reader: Box<dyn Read> = match matches.value_of("file") {
-        Some(filename) => Box::new(File::open(filename)?),
+        Some(filename) => {
+            let mut f = File::open(filename)?;
+
+            // apply start offset
+            if start_offset > 0 {
```

Comment:
Is there a reason that this is limited to files? I understand that `Stdin` doesn't implement `Seek`, but surely it'd be fairly easy to just ignore input from `StdinLock` by using [`BufRead`](https://doc.rust-lang.org/std/io/trait.BufRead.html)'s `fill_buf` and `consume`, 

One defense I can see for this is that it's not immediately obvious (to me) how to make that robust. See discussion below about errors for my thoughts on this.